lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2408,10 +2414,14 @@ postTransactionFeeOld ctx@ApiLayer{..} (ApiT walletId) body = do
     padding = Quantity 20
 
 constructTransaction
-    :: forall n
-     . HasSNetworkId n
-    => ApiLayer (SeqState n ShelleyKey) 'CredFromKeyK
-    -> ArgGenChange (SeqState n ShelleyKey)
+    :: forall n s k.
+        ( HasSNetworkId n
+        , s ~ SeqState n k
+        , k ~ ShelleyKey
+        , Excluding '[SharedKey] k
```

Comment:
🤔 The `~` and `Excluding` constraints would be overlapping conceptually, I think? `WalletFlavor s ~ ShelleyWallet` might suffice?

Reply:
you're right, removed unnecessary `Excluding`